### DISCUSSION TITLE
How can I stop a Task from inside of the Task?

### DISCUSSION THREAD

**Discussion by FloGamerMC** [Minestom/Minestom#880] [Help]
I would like to stop a Task after a variable have a specific value.

**Comment by MrGazdag** [comment]
Instead of `MinecraftServer.getSchedulerManager().buildTask()`, use `submitTask()`.

This method takes a `Supplier<TaskSchedule>`, which gets called whenever a task is run, and you return a `TaskSchedule` based on what you want your task to do. You can schedule the task again with `TaskSchedule.duration(Duration)` (or the likewise methods), `TaskSchedule.tick(int)`, or `TaskSchedule.nextTick()`. To stop a task, you can return `TaskSchedule.stop()`.
